### PR TITLE
automation of adding info into "GE Executions" google doc sheet

### PR DESCRIPTION
**Pull request by skaspi**
Related JIRA ticket:
 https://projects.engineering.redhat.com/browse/RHEVM-5311

### CODE REVIEW

File: vars/job_monitoring.groovy [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import groovy.json.JsonSlurper
+
+/**
+ * Extracts the displayName from given build.
+ *
+ * @param response String Object containing the response from  previous HTTP GET request.
+ * @param upstream_status String Object indicates whether this is an upstream build.
+ */
+def build_info(response, upstream_status=null) {
+    def object = null
```
Comment (kobihk) [inline vars/job_monitoring.groovy]: please don't use generic names for variables like "object"
you can use something like response_obj or whatever you want ;-)

File: src/rhvqe/CreateNewJob.groovy [hunk 2]
```diff
@@ -27,11 +27,11 @@ def createNewJob(args)
     def jobClusure = {
         _args ->
         def _jobMailBody =  "Scheduling ${_args.jobName} flow, \nplease follow ${_args.jobCurrentBuildUrl} to get more info."
-        notification.send_notification(
-          _args.jobMailSubject,
-          _jobMailBody,
-          _args.jobVersion
-        )
+//          notification.send_notification(
```
Comment (KKoukiou) [inline src/rhvqe/CreateNewJob.groovy]: If you don't want to execute this code - just delete it - don't comment it out.

File: vars/job_monitoring.groovy [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      }
+
+      def url = env.JENKINS_URL \
+             + object.actions.causes[0].upstreamUrl.join(", ") \
+             + object.actions.causes[0].upstreamBuild.join(", ") \
+             + "/api/json"
+
+      response = url.toURL().text
+      object = jsonSlurper.parseText(response)
+      } catch (Error e){}
```
Comment (KKoukiou) [inline vars/job_monitoring.groovy]: This here belongs to the try so it's not correctly indented. Delete two spaces pls.

Comment (KKoukiou) [inline vars/job_monitoring.groovy]: Catching Error is too generic. I don't really know groovy but the concept of exception handling is the same in every language, and the idea is that you should not catch too generic exceptions.
Please find the exception type that you need to catch and replace it with the generic Error.

File: vars/other.groovy [hunk 4]
```diff
@@ -92,3 +92,4 @@ def get_team_email(url) {
   }
   return "rhv-qe-devops@redhat.com"
 }
+
```
Comment (KKoukiou) [inline vars/other.groovy]: Unrelated newline change.

**Review by KKoukiou — COMMENTED** [review]
I took a 2 min look - some minor comments - I didn't review the functionality though ;)

File: src/rhvqe/CreateNewJob.groovy (hunk 2 above)
Comment (skaspi) [reply src/rhvqe/CreateNewJob.groovy]: It is only for debugging. After finishing, I will return it back

File: vars/job_monitoring.groovy (hunk 3 above)
Comment (skaspi) [reply vars/job_monitoring.groovy]: I will indent it properly, Which two spaces you mean,please?

Comment (KKoukiou) [reply vars/job_monitoring.groovy]: The catch line should be indented back to spaces to match the indentation of the try.
See example https://www.tutorialspoint.com/groovy/groovy_exception_handling.htm

File: vars/other.groovy (hunk 4 above)
Comment (skaspi) [reply vars/other.groovy]: Done

File: vars/job_monitoring.groovy (hunk 3 above)
Comment (skaspi) [reply vars/job_monitoring.groovy]: Ok

File: vars/job_monitoring.groovy [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import groovy.json.JsonSlurper
+
+/**
+ * Extracts the displayName from given build.
+ *
+ * @param response String Object containing the response from  previous HTTP GET request.
+ * @param upstream_status String Object indicates whether this is an upstream build.
```
Comment (kobihk) [inline vars/job_monitoring.groovy]: I think the term "upstream" can confuse so I would suggest to replace it with something like parent_pipeline_status or something more clear

File: vars/job_monitoring.groovy (hunk 1 above)
Comment (skaspi) [reply vars/job_monitoring.groovy]: Ok

File: vars/job_monitoring.groovy (hunk 5 above)
Comment (skaspi) [reply vars/job_monitoring.groovy]: Ok

File: vars/job_monitoring.groovy (hunk 3 above)
Comment (skaspi) [reply vars/job_monitoring.groovy]: After checking it more thoroughly, I came to decision to leave it as is.

Explanation: I agree with you that catching too broad exception is practice we should avoid. BUT, dealing with Jenkins pipelines written in Groovy, leads to unexpected behavior and time consuming debugging.

Catching board exceptions cover us from Groovy/Jenkins related unexpected behavior and pass all my tests so far.

So, with respect to all written above, I will leave it as is.

Comment (skaspi) [reply vars/job_monitoring.groovy]: What I mean by 'unexpected behavior' : sometimes the exception is thrown and sometimes not. When it is thrown, I don't understand its origin/cause. In addition, the types of these exceptions differ from time to time

File: vars/jobs.groovy [hunk 6]
```diff
@@ -14,7 +14,7 @@ def get_job_url(job){
 def build_job(name, parameters, is_stable)
 {
   def job = null
-  job = build job: name, parameters: parameters, propagate: false
+  job = build job: name, parameters: parameters, propagate: false, wait: true
```
Comment (kobihk) [inline vars/jobs.groovy:17]: with this wait we will not be able to run jobs in parallel am I right?

Comment (skaspi) [reply vars/jobs.groovy:17]: No, it has nothing common with parallel job execution.

**Review by kobihk — APPROVED** [review]
